### DISCUSSION TITLE
Event doesn't seem to adhere to pre/post capture setting

### DISCUSSION THREAD

**Discussion by bluesk1d** [blakeblackshear/frigate#11998] [Report a Bug] [answered]
### Describe the problem you are having

While reviewing the events in the Alerts or Detections tab, hovering over them displays a preview of the full event including pre and post capture duration. Clicking the event for full 1x preview results in playback starting (seemingly) at the moment of positive ID which can be several seconds "late". The extremely handy direct export button from the event review page becomes less useful when the exported event can cut off relevant portions of the clip.

### Steps to reproduce

On the review summary page, hover over any event. The full clip is previewed, including pre and post capture period.
Click on the clip. The clip does not contain the pre and post capture period.
Export the clip. The export does not contain the pre and post capture period.

### Version

0.14.0-4e5a6eb

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
NA
```

### Operating system

Other Linux

### Install method

Docker CLI

### Network connection

Wired

### Camera make and model

NA

**Comment by NickM-27** [comment]
This is not a bug and working as expected. The pre / post affects what is recorded, not what is shown in the UI

**Reply by bluesk1d** [reply]
Ah, I record 24x7 so I wouldn't notice a difference in what is committed to disk. 

Should that be the default behavior? Is there a situation where users would prefer the defined pre/post capture didn't apply to events and exports? The hover preview seems ideal. When reviewing the event, you often only see the activity some time after it started. It also makes the export button not useful since you need to go manually do an export to capture the beginning.

Maybe this could be configurable?

**Reply by ofcedwards** [reply]
I'll second this suggestion. I think it's logical, especially with exports, for users to expect their pre/post capture settings to be reflected in the saved clip. Without that, manually specifying the start and stop time of the export is necessary to retain the full clip that was saved. I think the purpose of pre/post capture is somewhat defeated by not including those seconds when right-clicking an event and selecting export. The docs also describe pre/post capture as number of seconds before/after the event to include, which I think would lead people to expect those seconds to be included in the event itself.

**Reply by NickM-27** [reply]
Events don't really exist in the UI anymore, it's now review items. Internally this still affects events but that is different than review items, so it would be something that needs to be thought through before this behavior changes, I wouldn't expect it to change for 0.14.0

**Reply by bluesk1d** [reply]
Yeah I'm probably using the incorrect 0.14 nomenclature when I say events. I'm starting to second guess my memory but didn't events in 0.13 include the pre/post roll?

**Reply by ofcedwards** [reply]
I think replacing the familiar "events" with "reviews" will likely have a lot of users confusing the terminology. Regardless, I think the point stands that users won't find it desirable to have an exported review item start/stop short just because object detection hadn't yet started or stopped a bit early. A buffer of 5 seconds on each side is typically enough to feel like the clip truly encompassed everything from start to finish. I know clicking a review item already takes users back two seconds from the actual start timestamp, so why not have it configurable? And regarding exports, why not have a config item much like pre/post capture that allows users to set a buffer on either side of review exports?

**Reply by NickM-27** [reply, marked as answer]
We don't want to have a bunch of small config items for ui behavior, that is difficult to maintain as we inevitably make changes to the ui in the future. I do think it makes sense to apply the review padding to the esports the same as it already is when opening a review item recording.

**Reply by ofcedwards** [reply]
That's fair enough. I can see the benefit of keeping configurable options down to a minimum. I think adding the review padding to the front/back of exports would likely be sufficient.

**Reply by bluesk1d** [reply]
FWIW, if there is padding already applied to the review item, it's often not enough. What padding is applied to the hover 8x preview? Whatever that is seems to be ideal.
